src/solver/sLR.ml

```diff
@@ -66,7 +66,7 @@ module SLR3 =
           if tracing then trace "sol" "Contrib:%a" S.Dom.pretty tmp;
           let tmp =
             if wpx then
-              if HM.mem globals x then S.Dom.widen old tmp
+              if HM.mem globals x then S.Dom.widen old tmp (* TODO: no join in second argument, can call widen incorrectly? *)
```

Comment:
Should we not fix this rather than produce TODOs?

Reply:
I wasn't sure if SLR3 maybe ensures it somehow indirectly when handling side effects per origin. I now ran all tests with `slr3` and it triggered some invalid widening calls, so I made the fix now.